src/DIRAC/Resources/IdProvider/OAuth2IdProvider.py

```diff
@@ -560,3 +560,27 @@ def getUserProfile(self):
         except Exception as e:
             self.log.exception(e)
             return S_ERROR("Cannot get user profile: %s" % repr(e))
+
+    def getTokenFromAgent(self, userName: str, requiredScope=None, requiredTimeLeft=3500):
```

Comment:
I thought we decided not to have any code for `oidc-agent`?

Reply:
I think this does not harm and can be useful without any extra charge